### PR TITLE
Add documentation for Connection Status Callback status and reason codes

### PR DESCRIPTION
**Pull request by ewertons**
# Checklist
- [x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/main/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [x] This pull-request is submitted against the `main` branch. 
  - [x] I have merged the latest `main` branch prior to submission and re-merged as needed after I took any feedback.
  - [x] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

There is no documentation on the status and reason codes for the Connection Status Callback function.

# Description of the solution
 
Added further details to the Connection and Messaging Reliability documentation explicitly explaining the Connection Status Callback status and reason codes.

### CODE REVIEW

File: doc/connection_and_messaging_reliability.md [hunk 1]
```diff
@@ -288,23 +288,68 @@ To access it the user can invoke one of the functions below, passing a callback
 
 On [iothub\_client](https://github.com/Azure/azure-iot-sdk-c/blob/2018-05-04/iothub_client/inc/iothub_client.h#L171) module:
 
+```c
 IOTHUB\_CLIENT\_RESULT IoTHubClient\_SetConnectionStatusCallback(IOTHUB\_CLIENT\_HANDLE iotHubClientHandle, IOTHUB\_CLIENT\_CONNECTION\_STATUS\_CALLBACK connectionStatusCallback, void\* userContextCallback);
```
Comment (vaavva) [inline doc/connection_and_messaging_reliability.md:292]:
```suggestion
IOTHUB_CLIENT_RESULT IoTHubClient_SetConnectionStatusCallback(IOTHUB_CLIENT_HANDLE iotHubClientHandle, IOTHUB_CLIENT_CONNECTION_STATUS_CALLBACK connectionStatusCallback, void* userContextCallback);
```
Formatting got a bit funky with the code formatting - I'm assuming the slashes aren't necessary!

File: doc/connection_and_messaging_reliability.md [hunk 2]
```diff
@@ -288,23 +288,68 @@ To access it the user can invoke one of the functions below, passing a callback
...
+```
 
 On [iothub\_client\_ll](https://github.com/Azure/azure-iot-sdk-c/blob/2018-05-04/iothub_client/inc/iothub_client_ll.h#L402) module:
 
+```c
 IOTHUB\_CLIENT\_RESULT IoTHubClient\_LL\_SetConnectionStatusCallback(IOTHUB\_CLIENT\_LL\_HANDLE iotHubClientHandle, IOTHUB\_CLIENT\_CONNECTION\_STATUS\_CALLBACK connectionStatusCallback, void\* userContextCallback);
```
Comment (vaavva) [inline doc/connection_and_messaging_reliability.md:298]:
```suggestion
IOTHUB_CLIENT_RESULT IoTHubClient_LL_SetConnectionStatusCallback(IOTHUB_CLIENT_LL_HANDLE iotHubClientHandle, IOTHUB_CLIENT_CONNECTION_STATUS_CALLBACK connectionStatusCallback, void* userContextCallback);
```

File: doc/connection_and_messaging_reliability.md [hunk 3]
```diff
@@ -288,23 +288,68 @@ To access it the user can invoke one of the functions below, passing a callback
...
 
 - When the device client gets disconnected due to any issues;
 
-> These include network availability and IoT hub connectivity issues, authentication failures.
+  > These include network availability and IoT hub connectivity issues, authentication failures.
 
 - When the device client ceases attempting to re-connect (if the Retry Policy no longer allows to).
 
-Please take a look [here](https://github.com/Azure/azure-iot-sdk-c/blob/2018-05-04/iothub_client/inc/iothub_client_ll.h#L144) for more info on the possible status and reasons.
+The type `IOTHUB\_CLIENT\_CONNECTION\_STATUS\_CALLBACK` is [defined](https://github.com/Azure/azure-iot-sdk-c/blob/2018-05-04/iothub_client/inc/iothub_client_ll.h#L183) as:
```
Comment (vaavva) [inline doc/connection_and_messaging_reliability.md:311]:
```suggestion
The type `IOTHUB_CLIENT_CONNECTION_STATUS_CALLBACK` is [defined](https://github.com/Azure/azure-iot-sdk-c/blob/2018-05-04/iothub_client/inc/iothub_client_ll.h#L183) as:
```

File: doc/connection_and_messaging_reliability.md [hunk 4]
```diff
@@ -288,23 +288,68 @@ To access it the user can invoke one of the functions below, passing a callback
...
+Indicates whether the device client is connected or not to the Azure IoT Hub.
+
+|Value|Description|
+|-|-|
+|IOTHUB_CLIENT_CONNECTION_UNAUTHENTICATED|Effectively means the device client is not ready to communicate with the Azure IoT Hub. The device client could be in any state from completely disconnected to not yet authenticated, including when brief disconnections occur for SAS token refreshes. See the list of IOTHUB_CLIENT_CONNECTION_STATUS_REASON status below for further details.|
+|IOTHUB_CLIENT_CONNECTION_AUTHENTICATED|The device client is **ready** to communicate with the Azure IoT Hub, being both connected and authenticated.|
+
+#### Understanding IOTHUB_CLIENT_CONNECTION_STATUS_REASON
+
+This enumeration attemps to provide a more specific reason for the current connection status of the device client. Its values depend on the transport protocol chosen by the user application for the Azure IoT C SDK client (AMQP, MQTT or HTTP) and on error granularity provided by the Azure IoT Hub.
```
Comment (vaavva) [inline doc/connection_and_messaging_reliability.md:332]:
```suggestion
This enumeration attempts to provide a more specific reason for the current connection status of the device client. Its values depend on the transport protocol chosen by the user application for the Azure IoT C SDK client (AMQP, MQTT or HTTP) and on error granularity provided by the Azure IoT Hub.
```

File: doc/connection_and_messaging_reliability.md [hunk 5]
```diff
@@ -288,23 +288,68 @@ To access it the user can invoke one of the functions below, passing a callback
...
+|IOTHUB_CLIENT_CONNECTION_EXPIRED_SAS_TOKEN|The SAS token used in the current [MQTT](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_mqtt_common.c#L2870) connection is expired and the client must reconnect with a new SAS token. This is an inplicit dependency on MQTT v3.1.1, which is not capable of refreshing authentication information in the same connection.|**Not applicable.** The AMQP protocol is capable of [refreshing authentication](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_amqp_cbs_auth.c#L572) within the same connection.|**Not applicable.** A [new SAS token is generated](https://github.com/Azure/azure-c-shared-utility/blob/master/src/httpapiexsas.c#L168) for each HTTP request sent to the Azure IoT Hub.|
+
+##### Azure IoT Hub Limitations on MQTT CONNECT Result Codes
+
+The Azure IoT Hub does not support all the MQTT CONNECT return code values defined in the [MQTT v3.1.1 specification](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Table_3.1_-), returning always `Not Authorized` (MQTT CONNECT Return Code 5) on MQTT CONNECT failure.
```
Comment (vaavva) [inline doc/connection_and_messaging_reliability.md:352]:
```suggestion
The Azure IoT Hub does not support all the MQTT CONNECT return code values defined in the [MQTT v3.1.1 specification](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Table_3.1_-), always returning `Not Authorized` (MQTT CONNECT Return Code 5) on MQTT CONNECT failure.
```

File: doc/connection_and_messaging_reliability.md [hunk 6]
```diff
@@ -288,23 +288,68 @@ To access it the user can invoke one of the functions below, passing a callback
...
+
+Please see a description of the values according to each transport protocol supported by the Azure IoT C SDK:
+
+|Value|MQTT|AMQP|HTTP|
+|-|-|-|-|
+|IOTHUB_CLIENT_CONNECTION_OK|The Azure IoT C SDK client is connected and ready to communicate with the Azure IoT Hub.|Same|Same|
+|IOTHUB_CLIENT_CONNECTION_COMMUNICATION_ERROR|If a telemetry message [times out receiving a PUBACK](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_mqtt_common.c#L2287) from the Azure IoT Hub, or if there is [an error](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_mqtt_common.c#L2269) sending a [PUBACK or PUBREC](https://github.com/Azure/azure-umqtt-c/blob/master/src/mqtt_client.c#L667) to Azure IoT Hub or if an [I/O error occurs when using WebSockets](https://github.com/Azure/azure-umqtt-c/blob/master/src/mqtt_client.c#L226).|If the [AMQP transport](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_amqp_common.c#306) encounters an authentication timeout, unexpected link DETACH from Azure IoT Hub, link ATTACH timeouts.|**Not applicable**.|
```
Comment (vaavva) [inline doc/connection_and_messaging_reliability.md:343]:
```suggestion
|IOTHUB_CLIENT_CONNECTION_COMMUNICATION_ERROR|If a telemetry message [times out receiving a PUBACK](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_mqtt_common.c#L2287) from the Azure IoT Hub, or if there is [an error](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_mqtt_common.c#L2269) sending a [PUBACK or PUBREC](https://github.com/Azure/azure-umqtt-c/blob/master/src/mqtt_client.c#L667) to Azure IoT Hub or if an [I/O error occurs when using WebSockets](https://github.com/Azure/azure-umqtt-c/blob/master/src/mqtt_client.c#L226).|If the [AMQP transport](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_amqp_common.c#L306) encounters an authentication timeout, unexpected link DETACH from Azure IoT Hub, link ATTACH timeouts.|**Not applicable**.|
```

File: doc/connection_and_messaging_reliability.md [hunk 7]
```diff
@@ -288,23 +288,68 @@ To access it the user can invoke one of the functions below, passing a callback
...
+|IOTHUB_CLIENT_CONNECTION_BAD_CREDENTIAL|**Not applicable.**. The MQTT transport does [map some MQTT CONNECT return code values](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_mqtt_common.c#L2116) to this status code, but these MQTT CONNECT return codes are not supported by the Azure IoT Hub. See [note](#azure-iot-hub-limitations-on-mqtt-connect-result-codes) below.|If a SAS-based authentication request [fails](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_amqp_common.c#L302). See Azure IoT Hub documentation on [device authentication](https://learn.microsoft.com/en-us/azure/iot-hub/iot-hub-dev-guide-sas?tabs=node#authenticating-a-device-to-iot-hub) for more details.|**Not applicable**.|
```
Comment (vaavva) [inline doc/connection_and_messaging_reliability.md:345]:
```suggestion
|IOTHUB_CLIENT_CONNECTION_BAD_CREDENTIAL|**Not applicable.** The MQTT transport does [map some MQTT CONNECT return code values](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_mqtt_common.c#L2116) to this status code, but these MQTT CONNECT return codes are not supported by the Azure IoT Hub. See [note](#azure-iot-hub-limitations-on-mqtt-connect-result-codes) below.|If a SAS-based authentication request [fails](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_amqp_common.c#L302). See Azure IoT Hub documentation on [device authentication](https://learn.microsoft.com/azure/iot-hub/iot-hub-dev-guide-sas?tabs=node#authenticating-a-device-to-iot-hub) for more details.|**Not applicable**.|
```
not sure how, but maybe there's a way to add a newline after **Not applicable** for this cell!

File: doc/connection_and_messaging_reliability.md [hunk 8]
```diff
@@ -288,23 +288,68 @@ To access it the user can invoke one of the functions below, passing a callback
...
+|IOTHUB_CLIENT_CONNECTION_DEVICE_DISABLED|Raised by the [MQTT transport](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_mqtt_common.c#L2114) if an MQTT CONNECT to the Azure IoT Hub is rejected. See [note](#azure-iot-hub-limitations-on-mqtt-connect-result-codes) below.|**Not applicable.**|**Not applicable.**|
+|IOTHUB_CLIENT_CONNECTION_RETRY_EXPIRED|The MQTT transport has made its [maximum number of attempts to reconnect](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_mqtt_common.c#L2818) to the Azure IoT Hub and it will not longer try.|The AMQP transport has made its [maximum number of attempts to reconnect](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_amqp_common.c#L267) to the Azure IoT Hub and it will not longer try.|**Not applicable**. Each new HTTP request sent to the Azure IoT Hub is done over a new HTTP connection.|
```
Comment (vaavva) [inline doc/connection_and_messaging_reliability.md:347]:
```suggestion
|IOTHUB_CLIENT_CONNECTION_RETRY_EXPIRED|The MQTT transport has made its [maximum number of attempts to reconnect](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_mqtt_common.c#L2818) to the Azure IoT Hub and it will no longer try.|The AMQP transport has made its [maximum number of attempts to reconnect](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_amqp_common.c#L267) to the Azure IoT Hub and it will no longer try.|**Not applicable**. Each new HTTP request sent to the Azure IoT Hub is done over a new HTTP connection.|
```

**Review by vaavva — APPROVED** [review]

File: doc/connection_and_messaging_reliability.md [hunk 9]
```diff
@@ -288,23 +288,68 @@ To access it the user can invoke one of the functions below, passing a callback
...
+|IOTHUB_CLIENT_CONNECTION_NO_NETWORK|If an MQTT CONNECT packet [fails to be sent](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_mqtt_common.c#L2264) to the Azure IoT Hub for any reason.|If the AMQP transport detects a [network connection issue](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_amqp_common.c#L720), which includes socket errors, AMQP ATTACH failures to CBS link (for authentication).|**Not applicable**.|
```
Comment (danewalton) [inline doc/connection_and_messaging_reliability.md:344]:
```suggestion
|IOTHUB_CLIENT_CONNECTION_NO_NETWORK|If an MQTT CONNECT packet [fails to be sent](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_mqtt_common.c#L2264) to the Azure IoT Hub for any reason.|If the AMQP transport detects a [network connection issue](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_amqp_common.c#L720), which includes socket errors, AMQP ATTACH fails to CBS link (for authentication).|**Not applicable**.|
```

File: doc/connection_and_messaging_reliability.md (hunk 8 above)
Comment (danewalton) [inline doc/connection_and_messaging_reliability.md:347]:
```suggestion
|IOTHUB_CLIENT_CONNECTION_RETRY_EXPIRED|The MQTT transport has made its [maximum number of attempts to reconnect](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_mqtt_common.c#L2818) to the Azure IoT Hub and it will no longer try.|The AMQP transport has made its [maximum number of attempts to reconnect](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_amqp_common.c#L267) to the Azure IoT Hub and it will no longer try.|**Not applicable**. Each new HTTP request sent to the Azure IoT Hub is done over a new HTTP connection.|
```

File: doc/connection_and_messaging_reliability.md [hunk 10]
```diff
@@ -288,23 +288,68 @@ To access it the user can invoke one of the functions below, passing a callback
...
+|IOTHUB_CLIENT_CONNECTION_EXPIRED_SAS_TOKEN|The SAS token used in the current [MQTT](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_mqtt_common.c#L2870) connection is expired and the client must reconnect with a new SAS token. This is an inplicit dependency on MQTT v3.1.1, which is not capable of refreshing authentication information in the same connection.|**Not applicable.** The AMQP protocol is capable of [refreshing authentication](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_amqp_cbs_auth.c#L572) within the same connection.|**Not applicable.** A [new SAS token is generated](https://github.com/Azure/azure-c-shared-utility/blob/master/src/httpapiexsas.c#L168) for each HTTP request sent to the Azure IoT Hub.|
```
Comment (danewalton) [inline doc/connection_and_messaging_reliability.md:348]:
```suggestion
|IOTHUB_CLIENT_CONNECTION_EXPIRED_SAS_TOKEN|The SAS token used in the current [MQTT](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_mqtt_common.c#L2870) connection is expired and the client must reconnect with a new SAS token. This is an implicit dependency on MQTT v3.1.1, which is not capable of refreshing authentication information in the same connection.|**Not applicable.** The AMQP protocol is capable of [refreshing authentication](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_amqp_cbs_auth.c#L572) within the same connection.|**Not applicable.** A [new SAS token is generated](https://github.com/Azure/azure-c-shared-utility/blob/master/src/httpapiexsas.c#L168) for each HTTP request sent to the Azure IoT Hub.|
```

**Review by danewalton — APPROVED** [review]

File: doc/connection_and_messaging_reliability.md (hunk 1 above)
Comment (ericwol-msft) [reply doc/connection_and_messaging_reliability.md:292]: Yes, that was needed before the ```c format specifier, please remove

File: doc/connection_and_messaging_reliability.md (hunk 6 above)
Comment (ericwol-msft) [inline doc/connection_and_messaging_reliability.md:343]:
```suggestion
|IOTHUB_CLIENT_CONNECTION_COMMUNICATION_ERROR|If a telemetry message [times out receiving a PUBACK](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_mqtt_common.c#L2287) from the Azure IoT Hub, or if there is [an error](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_mqtt_common.c#L2269) sending a [PUBACK or PUBREC](https://github.com/Azure/azure-umqtt-c/blob/master/src/mqtt_client.c#L667) to Azure IoT Hub or if an [I/O error occurs when using WebSockets](https://github.com/Azure/azure-umqtt-c/blob/master/src/mqtt_client.c#L226).|If the [AMQP transport](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_amqp_common.c#306) encounters an authentication timeout, unexpected link DETACH from Azure IoT Hub, or link ATTACH timeouts.|**Not applicable**.|
```

**Review by ericwol-msft — APPROVED** [review]

File: doc/connection_and_messaging_reliability.md [hunk 11]
```diff
@@ -288,23 +288,68 @@ To access it the user can invoke one of the functions below, passing a callback
...
 
-Please take a look [here](https://github.com/Azure/azure-iot-sdk-c/blob/2018-05-04/iothub_client/inc/iothub_client_ll.h#L144) for more info on the possible status and reasons.
+The type `IOTHUB\_CLIENT\_CONNECTION\_STATUS\_CALLBACK` is [defined](https://github.com/Azure/azure-iot-sdk-c/blob/2018-05-04/iothub_client/inc/iothub_client_ll.h#L183) as:
+```c
+typedef void(*IOTHUB_CLIENT_CONNECTION_STATUS_CALLBACK)(IOTHUB_CLIENT_CONNECTION_STATUS result, IOTHUB_CLIENT_CONNECTION_STATUS_REASON reason, void* userContextCallback);
+```
+
+The user application must provide a function that matches the function pointer definition above to provide it as an argument to any of the `*_SetConnectionStatusCallback` functions above.
+
+For reference, the connection status and reason enumerations (`IOTHUB_CLIENT_CONNECTION_STATUS` and `IOTHUB_CLIENT_CONNECTION_STATUS_REASON`, respectively) are defined in [iothub_client_core_common.h](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/inc/iothub_client_core_common.h#L143).
```
Comment (jspaith) [inline doc/connection_and_messaging_reliability.md:318]: Linking to the line number directly will be a problem if header ever changes.  Also the MU_DEFINE_ENUM stuff is hard to follow for someone new to SDK in any case.

I actually think the best docs for this are the ones you've written lines below in "#### Understanding IOTHUB_CLIENT_CONNECTION_STATUS".  So I think just pointing at header is enough.

File: doc/connection_and_messaging_reliability.md (hunk 3 above)
Comment (jspaith) [reply doc/connection_and_messaging_reliability.md:311]: I have similar feedback below -- the link above to `https://github.com/Azure/azure-iot-sdk-c/blob/2018-05-04/iothub_client/inc/iothub_client_ll.h#L183` - this won't change, but pointing to very old copy of SDK.

I think its best to just point to the header file but not line #'s.  If someone needs more than what docs have, they can easily search header for it.

File: doc/connection_and_messaging_reliability.md (hunk 4 above)
Comment (jspaith) [inline doc/connection_and_messaging_reliability.md:332]: "attempts to provide" - I think we can just say "provide".

File: doc/connection_and_messaging_reliability.md (hunk 6 above)
Comment (jspaith) [reply doc/connection_and_messaging_reliability.md:343]: Same concerns hard-coding line numbers to headers.  I do like the level of detail about what in protocol would cause a specific error to get sent up, since this is non-obvious just the way we mapped multiple protocols into same error name space.

I see other line number references below, I won't keep calling out but same general feedback.  

Although when we're linking to implementation files - like src/iothubtransport_mqtt_common.c - we don't even need to link to the implementation C file.  Customers can and do go this deep if they want, but if they get blocked they can also open GitHub issues and we'll help them.

**Review by jspaith — APPROVED** [review]
Thanks a lot for all this cleanup.

File: doc/connection_and_messaging_reliability.md (hunk 3 above)
Comment (ewertons) [reply doc/connection_and_messaging_reliability.md:311]: Oh, that's too bad. It's so neat to point to the exact line in the code when we reference it. I agree we need to get rid of the references to old releases (branch/tag 2018-05-04), but I will keep the line references. I don't expect them to shift that much  anyway, because we don't have much code churn in the C SDK anymore..

Comment (ewertons) [reply doc/connection_and_messaging_reliability.md:311]: Removed all references to the old branch, and pointed to `main`.

File: doc/connection_and_messaging_reliability.md (hunk 4 above)
Comment (ewertons) [reply doc/connection_and_messaging_reliability.md:332]: Fixed.

File: doc/connection_and_messaging_reliability.md (hunk 5 above)
Comment (ewertons) [reply doc/connection_and_messaging_reliability.md:352]: Done.

File: doc/connection_and_messaging_reliability.md (hunk 6 above)
Comment (ewertons) [reply doc/connection_and_messaging_reliability.md:343]: I will keep the references for now. It's neat to be able to reference the line number if needed.

File: doc/connection_and_messaging_reliability.md (hunk 7 above)
Comment (ewertons) [reply doc/connection_and_messaging_reliability.md:345]: There is, but it mixes html with the markdown, and we usually try to avoid that.

Comment (ewertons) [reply doc/connection_and_messaging_reliability.md:345]: Fixed the link

File: doc/connection_and_messaging_reliability.md (hunk 8 above)
Comment (ewertons) [reply doc/connection_and_messaging_reliability.md:347]: Fixed.

File: doc/connection_and_messaging_reliability.md (hunk 9 above)
Comment (ewertons) [reply doc/connection_and_messaging_reliability.md:344]: Changed to `failures to AMQP ATTACH to CBS link`

File: doc/connection_and_messaging_reliability.md (hunk 4 above)
Comment (ewertons) [reply doc/connection_and_messaging_reliability.md:332]: Well, it doesn't provide completely... I think it's more honest to say `attempts`, but I will simplify and change as you suggested.

Comment (ewertons) [reply doc/connection_and_messaging_reliability.md:332]: Fixed.

File: doc/connection_and_messaging_reliability.md [hunk 12]
```diff
@@ -288,23 +288,68 @@ To access it the user can invoke one of the functions below, passing a callback
...
+
+
+#### Understanding IOTHUB_CLIENT_CONNECTION_STATUS
+
+Indicates whether the device client is connected or not to the Azure IoT Hub.
```
[the remark this answers is not in the document]
Comment (ewertons) [reply doc/connection_and_messaging_reliability.md:323]: Fixed.

File: doc/connection_and_messaging_reliability.md (hunk 2 above)
Comment (ewertons) [reply doc/connection_and_messaging_reliability.md:298]: Fixed.

File: doc/connection_and_messaging_reliability.md (hunk 6 above)
Comment (vaavva) [reply doc/connection_and_messaging_reliability.md]: This change was because the link didn't go to the line properly, it needs the `L`

File: doc/connection_and_messaging_reliability.md [hunk 13]
```diff
@@ -286,48 +290,97 @@ The Azure IoT Device Client C SDK provides a callback option to notify the upper
...
+|IOTHUB_CLIENT_CONNECTION_BAD_CREDENTIAL|**Not applicable.**. The MQTT transport does [map some MQTT CONNECT return code values](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_mqtt_common.c#L2116) to this status code, but these MQTT CONNECT return codes are not supported by the Azure IoT Hub. See [note](#azure-iot-hub-limitations-on-mqtt-connect-result-codes) below.|If a SAS-based authentication request [fails](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_amqp_common.c#L302). See Azure IoT Hub documentation on [device authentication](https://learn.microsoft.com/azure/iot-hub/iot-hub-dev-guide-sas?tabs=node#authenticating-a-device-to-iot-hub) for more details.|**Not applicable**.|
```
Comment (vaavva) [inline doc/connection_and_messaging_reliability.md:349]:
```suggestion
|IOTHUB_CLIENT_CONNECTION_BAD_CREDENTIAL|**Not applicable.** The MQTT transport does [map some MQTT CONNECT return code values](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_mqtt_common.c#L2116) to this status code, but these MQTT CONNECT return codes are not supported by the Azure IoT Hub. See [note](#azure-iot-hub-limitations-on-mqtt-connect-result-codes) below.|If a SAS-based authentication request [fails](https://github.com/Azure/azure-iot-sdk-c/blob/main/iothub_client/src/iothubtransport_amqp_common.c#L302). See Azure IoT Hub documentation on [device authentication](https://learn.microsoft.com/azure/iot-hub/iot-hub-dev-guide-sas?tabs=node#authenticating-a-device-to-iot-hub) for more details.|**Not applicable**.|
```
